compiler/dalgona/README.md

```diff
@@ -0,0 +1,72 @@
+# dalgona
+
+## What is dalgona?
+
+`dalgona` is a tool for dynamic analysis of deep neural network.
+
+## How it works?
+
+`dalgona` runs a user's custom analysis code (written in "Python") while performing inference. The analysis code has the form of hooks, called before/after each operator is executed. Intermediate execution results (values of activations) are passed to the hooks, so users can analyze the distribution of activations inside the hooks. The analysis result can be exported as files, log messages or any other forms, used for various purposes (model compression, optimization, etc.).
+
+## Possible applications
+- Finding quantization parameters based on the distribution of activations
+- Finding sparse activations by observing the portion of zero values
+- Finding the distribution of conditional variables in If-statement and While-statement
+- Visualization of activation data with Python libraries
+
+## Prerequisite
+- Python (apt install python-dev python3-dev)
+- Circle model (target to analyze)
+- Input data of the model (hdf5 format. See imgdata2hdf5 or gen_h5_explicit_inputs.py for more details.)
+- Analysis code (Python code)
+
+## Example
+```
+dalgona \
+> --input_model model.circle
+> --input_data data.hdf5
+> --analysis analysis/AnalysisTemplate.py
+```
+
+## Arguments
+```
+  --help            Show help message and exit
+  --input_model     Input model filepath (.circle)
+  --input_data      Input data filepath (.h5)
+  --analysis        Analysis code filepath (.py)
+  --analysis_args   (optional) String argument passed to the analysis code
+```
+
+## How to write analysis code?
+
+`dalgona` provides hooks which are called before/after an operator is executed.
+Users can access tensors relevant to the corresponding operator inside the hooks.
+The information of each operator is passed as the arguments of the hook.
+For example, for a Conv2D operator, dalgona provides following hooks.
```

Comment:
We usually used our sub-project or module names as italic, _dalgona_ . You use `dalgona` and dalgona. It would be better to use one convention.